examples/testdata/wasi_args.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+//go:wasm-module wasi_snapshot_preview1
+//export args_get
+func args_get(uint32, uint32) uint16
+
+//go:wasm-module wasi_snapshot_preview1
+//export args_sizes_get
+func args_sizes_get(uint32, uint32) uint16
+func main() {
+	testGetWASIArgs()
+	testInvalidArguments()
+}
+
+func testGetWASIArgs() {
+	var expectedArgs = []string{
+		"foo",
+		"bar",
+		"foobar",
+		"",
+		"baz",
+	}
+
+	if len(os.Args) != len(expectedArgs) {
```

Comment:
basically verify this by checking console in the unit tests, which reduces code also as you can use testify.

Reply:
Yeah checking the output from the guest in the host is one way to check the expected conditions. However, it works well for simple system calls like this, but I guess it's easier to read if the comparison gets complicated such as comparing the expected struct with the returned value. Indeed we cannot use testify in the guest, though. What do you think?

Reply:
if we are interested in grey box testing, it would be not in an example, maybe we can put something in the memory section and verify it later, if the aim is also to use the start component. generally I think it is conflating as it isn't required to use the start component to reach the values (we can export the function and call it directly)